kombu/asynchronous/aws/connection.py

```diff
@@ -92,7 +92,8 @@ def getrequest(self):
         headers = Headers(self.headers)
         return self.Request(self.path, method=self.method, headers=headers,
                             body=self.body, connect_timeout=self.timeout,
-                            request_timeout=self.timeout, validate_cert=False)
+                            request_timeout=self.timeout,
+                            validate_cert=True, ca_certs=get_cert_path(True))
```

Comment:
the issue is a bit more complex and multi-layered. i am going to lay out my timeline and go a bit deeper into my reasoning.

- i implemented urllib3 Pool with default settings. and totally missed all the extra `Request.*` parameters. and no proxy support
- then i implemented proxy support and the missed `Request` parameters. and then i started to test these changes on my aws environment (not production, so i can break it)

with the first implementation of parameters i started to get these urllib3 warnings:
```
[2024-10-08 15:14:15,172: WARNING/MainProcess] /var/app/venv/staging-LQM1lest/lib64/python3.11/site-packages/urllib3/connectionpool.py:1099: InsecureRequestWarning: Unverified HTTPS request is being made to host 'sqs.eu-west-1.amazonaws.com'. Adding certificate verification is strongly advised. See: https://urllib3.readthedocs.io/en/latest/advanced-usage.html#tls-warnings
```
and i don't like Warnings,.. and they are related to my new code, so i started investigating.

at first i saw the default value on class `Request` `validate_cert = True` in `http/base.py`. so i thought the problem is somewhere else... 

then i saw that in `aws/connection.py` the `validate_cert=` is set to `False`. so i changed this to `True`(i mentioned it here in comments)... 

then i started to get other errors, that complain about `ca_certs`. this issue was my mistake(1 extra bad line of config). but i fixed in a more proper/better way:

As we already have a `botocore` dependency, and it relies on urllib3 - i decided to see how they handle the connection.
they use `get_cert_path` method. which at first tries to use `certifi.where` and if `certifi` is not present - use their own pre-bundled `cacert.pem`.

here in `kombu` we have 2 layers: 1st in `http/base` for general use; and 2nd in `aws/connection` for AWS use on top of base.
then i split the fix into 2 layers:
- if there is no "custom" certificate in the `Request` - `http/base` will first try to find `ceritifi`(which doesn't require `certifi` dependency), and if not present - keep the default `urllib3.ca_certs` value (i.e. not override with anything - use default). and if there is a custom `Request.ca_certs` - it will set it. resulting in: (optional) `certifi` dependency and no `botocore` dependency
- and on `aws/connection`, as we already depend on `botocore` - i decided to import and use `ca_certs=get_cert_path(True)` with the default `botocore` behaviour. which is a bit more advanced than in `http/base`

and only afterwards i found the recent Issues and changes, that Disable certificate validation for aws connection. 
https://github.com/celery/kombu/issues/2113 and https://github.com/celery/kombu/pull/2114
but they were about `pycurl` which we don't heave anymore ;) and i think it is not wise to reduce security level , just to hotfix an issue.

technically, this PR is 2 PRs:
- replace `pycurl` with `urllib3`
- re-enable ssl verification for AWS requests (now on urllib3 as in botocore and also without it)

Reply:
Wow. What a great explanation!
Thank you very much, great great work!! 👏